src/linear_algebra/affine_space/affine_map.lean

```diff
@@ -0,0 +1,468 @@
+/-
+Copyright (c) 2020 Joseph Myers. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Joseph Myers.
+-/
+import algebra.add_torsor
+import linear_algebra.tensor_product
+import data.set.intervals.unordered_interval
+
+/-!
+# Affine maps
+
+This file defines affine maps.
+
+## Main definitions
+
+* `affine_space V P` is a notation for `add_torsor V P` in the case of `module k V`. `P` is the type
+  of points in the space and `V` the `k`-module of displacement vectors.  Definitions and results
+  not depending on the `module` structure appear in `algebra.add_torsor` instead of here; that
+  includes the instance of an `add_group` as an `add_torsor` over itself, which thus gives a
+  `module` as an `affine_space` over itself.  Definitions of affine spaces vary as to whether a
+  space with no points is permitted; here, we require a nonempty type of points (via the definition
+  of torsors requiring a nonempty type).  Affine spaces are defined over any module, with stronger
+  type class requirements on `k` being used for individual lemmas where needed.
```

Comment:
Since this file isn't the place where `affine_space` is defined, maybe this sort of general comment should go in `algebra.add_torsor` instead as the place that defines that notation; this PR doesn't seem to leave any better place for general documentation about affine spaces, unless you create a `default.lean` just to have such doc strings. Likewise in the implementation notes below.